proposals/0027-content-schemas.md

```diff
@@ -0,0 +1,546 @@
+  image: z.string().optional(),
+  tags: z.array(z.string()),
+  // transform to another data type with `transform`
+  // ex. convert date strings to Date objects
+  publishedDate: z.string().transform((str) => new Date(str)),
+});
+```
+
+Let’s break these features down.
+
+# Detailed usage
+
+As you might imagine, Content Schemas have a lot of moving parts. Let's detail each one:
+
+## The `src/content/` directory
+
+This RFC introduces a new, reserved directory for Astro to manage: `{srcDir}/content/`. This directory is where all collections and schema definitions live, relative to your configured source directory.
+
+## The `.astro` cache directory
+
+Since we will preprocess your post frontmatter separate from Vite ([see background](#background)), we need a new home for generated metadata. This will be a special `.astro` directory **generated by Astro** at build time or on dev server startup.
+
+The user is *not* expected to view or edit this manifest. This only exists to enable type checking and frontmatter parsing via `getCollection` and `getEntry`.
+
+From the user's perspective, `.astro` should be the import source for all *generated* helpers that smartly understand your project.  Today, this would include [`getCollection` and `getEntry`](#example) imported like so:
+
+```tsx
+import { getCollection, getEntry } from '.astro';
+```
+
+We expect `.astro` to live at the base of your project directory. This falls inline with generated directories like `.vscode` for editor tooling and `.vercel` for deployments. We will preconfigure a type alias as well, allowing users to import `.astro` instead of `../../.astro`.
+
+## Creating a collection
+
+All entries in `src/content/` **must** be nested in a "collection" directory. This allows you to group content based on the schema their frontmatter should use. This is similar to creating a new table in a database, or a new content model in a CMS like Contentful.
+
+What this looks like in practice:
+
+```bash
+src/content/
+  newsletters/
+    # All newsletters have the same frontmatter properties
+    ~schema.ts
+    week-1.md
+    week-2.md
+    week-3.md
+  blog/
+    # All blog posts have the same frontmatter properties
+    ~schema.ts
+    columbia.md
+    enterprise.md
+    endeavour.md
+```
+
+### Nested directories
+
+Collections are considered **one level deep**, so you cannot nest collections or schemas within other collections. However, we *will* allow nested directories to better organize your content. This is vital for certain use cases like internationalization:
+
+```bash
+src/content/
+  # Applies to all nested directories 👇
+  ~schema.ts
+  docs/
+    en/
+    es/
+    ...
+```
+
+All nested directories will share the same schema defined at the top level. Which brings us to...
+
+## Adding a schema
+
+To add type checking to a given collection, you can add a `~schema.{js|mjs|ts}` file inside of that collection directory. This file should:
```

Comment:
From the call, schemas are not required and you can have collections that do not have them, in which case they will be typed as `any`. This should be explained somewhere in the document.